### PR TITLE
Split z64.h: gfxprint, message, pause, view (as new headers), +misc

### PR DESCRIPTION
**Pull request by Dragorn421**
-> gfxprint, message, pause, view

### CODE REVIEW

File: include/z64pause.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#ifndef Z64PAUSE_H
```
Comment (engineer124) [inline include/z64pause.h:1]: Would it be more appropriate to put this header in:
`src/overlays/misc/ovl_kaleido_scope/pause.h`
similar to how there is:
`src/overlays/gamestates/ovl_file_choose/file_select.h`

Comment (Dragorn421) [reply include/z64pause.h:1]: I plan to have this header used for "public" pause stuff that is used in code, and the one in overlays/ for stuff that only matters inside kaleido

It's not always a clear separation but 🤷

Comment (fig02) [reply include/z64pause.h:1]: yea file select gets away with being more isolated cause it is its own gamestate. not so much for pause

File: include/z64gfxprint.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#ifndef Z64GFXPRINT_H
```
Comment (Thar0) [inline include/z64gfxprint.h:1]: This header could probably already do without the z64 prefix

File: include/z64gfxprint.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#define Z64GFXPRINT_H
+
+#include "z64.h"
```
Comment (Thar0) [inline include/z64gfxprint.h:4]: Why include `z64.h`? Including `ultra64.h` and `color.h` seems sufficient?

File: include/z64pause.h [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+#define Z64PAUSE_H
+
+#include "z64.h"
```
Comment (Thar0) [inline include/z64pause.h:4]: `z64animation.h` and `z64message.h` instead?

File: include/z64view.h [hunk 5]
```diff
@@ -0,0 +1,52 @@
+#ifndef Z64VIEW_H
+#define Z64VIEW_H
+
+#include "z64.h"
```
Comment (Thar0) [inline include/z64view.h:4]: `z64math.h` instead?

File: include/z64pause.h [hunk 6]
```diff
@@ -0,0 +1,135 @@
+#ifndef Z64PAUSE_H
+#define Z64PAUSE_H
+
+#include "z64animation.h"
+#include "z64message.h"
+#include "z64view.h"
+
```
Comment (engineer124) [inline include/z64pause.h:7]:
```suggestion
#include "z64animation.h"
#include "z64message.h"
#include "z64ocarina.h"
#include "z64view.h"
```
Would also be worth adding `z64ocarina.h` in `z64message.h` as both these headers have an `OcarinaStaff*` struct member in them

**Comment by Dragorn421** [comment]
To comment back on Tharo's review, I'm very liberally splitting headers for the sake of having them split at all,
I don't have an opinion at this point about z64-prefixing or not (I have the opinion that prefix will go eventually though)
and "stricter" includes are good if we bother to, but as long as it compiles I'm fine with one-giant-header-but-split-in-several-files, to me this is all little steps and it'll need further cleanup

tldr: thanks for the review, it is very welcome & helpful, the answers to "why" is "I don't care about it yet"

File: include/z64pause.h (hunk 6 above)
Comment (Dragorn421) [reply include/z64pause.h:7]: `OcarinaStaff` being only used for pointers I used a forward declaration of the struct instead

**Review by hensldm — APPROVED** [review]

File: include/z64pause.h [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+struct OcarinaStaff;
+
+typedef enum {
+    /* 0 */ PAUSE_BG_PRERENDER_OFF, // Inactive, do nothing.
+    /* 1 */ PAUSE_BG_PRERENDER_SETUP, // The current frame is only drawn for the purpose of serving as the pause background.
+    /* 2 */ PAUSE_BG_PRERENDER_PROCESS, // The previous frame was PAUSE_BG_PRERENDER_SETUP, now apply prerender filters.
+    /* 3 */ PAUSE_BG_PRERENDER_DONE, // The pause background is ready to be used.
+    /* 4 */ PAUSE_BG_PRERENDER_MAX
+} PauseBgPreRenderState;
```
Comment (fig02) [inline include/z64pause.h:16]: something tells me this sorta belongs to z_play more than anything else, but I understand why it makes sense here too. But the different states are related more to how z_play should processing a drawing frame more than anything 

curious on other thoughts, if the concensus is keep it here then thats fine

Comment (Dragorn421) [reply include/z64pause.h:16]: Yeah I guess the implementation is more z_play-ish than pause-ish, I just grabbed it since "hey it has pause in the name". I'll move it (comments on (not) moving it still welcome)

**Review by Thar0 — APPROVED** [review]

File: include/z64pause.h (hunk 7 above)
Comment (Dragorn421) [reply include/z64pause.h:16]: Wait I thought for some reason "z_play" had its own header but it doesn't

so instead of leaving this in z64.h for now I think it's better in z64pause.h

Comment (hensldm) [reply include/z64pause.h:16]: Small note, but whatever happens to `PauseBgPreRenderState` the same thing should probably happen to `TransitionTileStatus` (after it gets merged of course :) )

Comment (Dragorn421) [reply include/z64pause.h:16]: > Small note, but whatever happens to `PauseBgPreRenderState` the same thing should probably happen to `TransitionTileStatus` (after it gets merged of course :) )

wdym? Move `TransitionTileStatus` to z64pause.h too if we do leave `PauseBgPreRenderState` in z64pause.h ?
Or move `TransitionTileStatus` to z64.h (with `PlayState`) if we do leave `PauseBgPreRenderState` in z64.h ?

Comment (hensldm) [reply include/z64pause.h:16]: If `PauseBgPreRenderState` stays in z64pause.h , then `TransitionTileStatus` should probably stay in z64transition (or possibly `fbdemo/TransitionTile` gets its own header eventually).  If we do leave `PauseBgPreRenderState` in z64.h then `TransitionTileStatus` should probably also go in z64.h with `PlayState`

Comment (fig02) [reply include/z64pause.h:16]: `so instead of leaving this in z64.h for now I think it's better in z64pause.h`
I dont really agree, it staying in z64.h just means it doesnt have a home yet, which is true (until play is properly split)

Comment (Dragorn421) [reply include/z64pause.h]: Alrighty I moved both enums (pausebg and transitiontile) near PlayState

**Review by fig02 — APPROVED** [review]
